frontend/src/lib/hooks/useScreenSize.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+import { useTheme } from 'styled-components';
+
+export const useScreenSize = () => {
+  const {
+    breakpoints: { S, M },
+  } = useTheme();
+
+  const [screenSize, setScreenSize] = useState({
+    isLarge: window.innerWidth > M,
+    isMedium: window.innerWidth >= S && window.innerWidth <= M,
+  });
+
+  const checkScreenSize = () => {
+    setScreenSize({
+      isLarge: window.innerWidth > M,
+      isMedium: window.innerWidth >= S && window.innerWidth <= M,
+    });
+  };
+
+  useEffect(() => {
+    window.addEventListener('resize', checkScreenSize);
```

Comment:
maybe we can write the function exactly in the `useEffect` so we don't have to lie about the dependencies for the `useEffect`.